### DISCUSSION TITLE
"accept" button can't add the items to the categories in the cookie value

### DISCUSSION THREAD

**Discussion by Tan-Wenhao** [orestbida/cookieconsent#621] [Q&A]
The `categories` always show empty array even though clicking the "accept all" button. Anything wrong?

`{"categories":[],"revision":0,"data":null,"consentTimestamp":"2024-01-22T11:55:01.670Z","consentId":"05cac639-b342-4a5e-a250-6de6e61502a5","services":{"necessary":[],"analytics":[],"ads":[]},"lastConsentTimestamp":"2024-01-22T11:55:01.670Z","expirationTime":1721649301671}`

**Comment by orestbida** [comment]
@Wenhao-Tan please provide your config. or a demo

**Reply by Tan-Wenhao** [reply]
`CookieConsent.run({

    categories: {
        necessary: {
            readOnly: true,
            enabled: true
        },
        analytics: {
            autoClear: {
                cookies: [
                    {
                        name: /^(_ga|_gid)/
                    }
                ]
            }
        },
        ads: {}
    },

    language: {
        default: 'en',
        translations: {
            en: {
                consentModal: {
                    title: 'We use cookies',
                    description: 'Cookie modal description',
                    acceptAllBtn: 'Accept all',
                    acceptNecessaryBtn: 'Reject all',
                    showPreferencesBtn: 'Manage Individual preferences'
                },
                preferencesModal: {
                    title: 'Manage cookie preferences',
                    acceptAllBtn: 'Accept all',
                    acceptNecessaryBtn: 'Reject all',
                    savePreferencesBtn: 'Accept current selection',
                    closeIconLabel: 'Close modal',
                    sections: [
                        {
                            title: 'Somebody said ... cookies?',
                            description: 'I want one!'
                        },
                        {
                            title: 'Strictly Necessary cookies',
                            description: 'These cookies are essential for the proper functioning of the website and cannot be disabled.',
                            //this field will generate a toggle linked to the 'necessary' category
                            linkedCategory: 'necessary'
                        },
                        {
                            title: 'Performance and Analytics',
                            description: 'These cookies collect information about how you use our website. All of the data is anonymized and cannot be used to identify you.',
                            linkedCategory: 'analytics',
                        },
                        {
                            title: 'More information',
                            description: 'For any queries in relation to my policy on cookies and your choices, please <a href="#contact-page">contact us</a>'
                        }
                    ]
                }
            }
        }
    }
});`

**Reply by Tan-Wenhao** [reply]
It's working in a brand new project, but not working in my current project. What factors could prevent it from adding the category to the array?

**Reply by orestbida** [reply]
it works just fine on my end:

```json
{
   "categories": [
      "necessary",
      "analytics",
      "ads"
   ],
   "revision": 0,
   "data": null,
   "consentTimestamp": "2024-01-22T12:43:58.195Z",
   "consentId": "385eac7b-b5d5-4ded-87ca-53ca3f22f88e",
   "services": {
      "necessary": [],
      "analytics": [],
      "ads": []
   },
   "lastConsentTimestamp": "2024-01-22T12:43:58.195Z",
   "expirationTime": 1721652238195
}
```

Are you using a web server (address should start with "http:// or https://") ?

**Reply by orestbida** [reply]
> It's working in a brand new project, but not working in my current project. What factors could prevent it from adding the category to the array?

I'm not sure, I would need to take a look into the project itself.

**Reply by Tan-Wenhao** [reply]
Working with v2.9.1 on my current project, I have to switch it back to the old version.

**Reply by orestbida** [reply]
The only thing that comes to mind is the fact that v3 enforces an rfcCompliant cookie. You could also try enabling the ` cookie.useLocalStorage` option to store the value in localstorage instead.
